libs/designsystem/src/lib/components/checkbox/checkbox.component.spec.ts

```diff
@@ -1,105 +1,125 @@
     });
 
-    it('should emit false', () => {
-      spyOn(spectator.component.checkedChange, 'emit');
-      const change = false;
+    it('should have class .attention-level1 when attentionLevel is 1', () => {
+      const attentionLevel = '1';
 
-      spectator.component.onChecked(change);
+      spectator.setInput('attentionLevel', attentionLevel);
+      spectator.detectChanges();
 
-      expect(spectator.component.checkedChange.emit).toHaveBeenCalledTimes(1);
-      expect(spectator.component.checkedChange.emit).toHaveBeenCalledWith(false);
+      expect(spectator.element).toHaveClass('attention-level1');
     });
-  });
 
-  describe('Inputs', () => {
-    it('should bind the class .square to ion-checkbox', () => {
-      const squareShape = 'square';
-      spectator.component.ngOnChanges({
-        shape: new SimpleChange(null, squareShape, false),
-      });
+    it('should set the [checked] input on ion-checkbox to true', () => {
+      const checked = true;
 
-      spectator.setInput('shape', squareShape);
+      spectator.setInput('checked', checked);
 
-      expect(spectator.query('ion-checkbox')).toHaveClass('square');
-      expect(spectator.query('ion-checkbox')).not.toHaveClass('circle');
+      expect(checkbox.checked).toBe(true);
     });
 
-    it('should bind the class .circle to ion-checkbox', () => {
-      const circleShape = 'circle';
-      spectator.component.ngOnChanges({
-        shape: new SimpleChange(null, circleShape, false),
-      });
+    it('should set the [checked] input on ion-checkbox to false', () => {
+      const checked = false;
 
-      spectator.setInput('shape', circleShape);
+      spectator.setInput('checked', checked);
 
-      expect(spectator.query('ion-checkbox')).toHaveClass('circle');
-      expect(spectator.query('ion-checkbox')).not.toHaveClass('square');
+      expect(checkbox.checked).toBe(false);
     });
 
-    it('should bind the class .primary to ion-checkbox', () => {
-      const color = 'primary';
-      spectator.component.ngOnChanges({
-        color: new SimpleChange(null, color, false),
-      });
+    it('should set the [disabled] input on ion-checkbox to true', () => {
+      const disabled = true;
 
-      spectator.setInput('color', color);
+      spectator.setInput('disabled', disabled);
 
-      expect(spectator.query('ion-checkbox')).toHaveClass('primary');
-      expect(spectator.query('ion-checkbox')).not.toHaveClass('secondary');
+      expect(checkbox.disabled).toBe(true);
     });
 
-    it('should set the [checked] input on ion-checkbox to true', () => {
+    it('should set the [disabled] input on ion-checkbox to false', () => {
+      const disabled = false;
+
+      spectator.setInput('disabled', disabled);
+
+      expect(checkbox.disabled).toBe(false);
+    });
+
+    it('should have error class when [hasError] input is true', () => {
+      const hasError = true;
+
+      spectator.setInput('hasError', hasError);
+      spectator.detectChanges();
```

Comment:
Consider inlining for readability
```suggestion
      spectator.setInput('hasError', true);
      spectator.detectChanges();
```